pkg/broker/ingress/options.go

```diff
@@ -40,28 +40,41 @@ func WithPort(port int) HandlerOption {
 func WithProjectID(id string) HandlerOption {
 	return func(h *handler) error {
 		ctx := context.Background()
-		client, err := newPubSubClient(ctx, id)
+		p, err := pubsub.NewClient(ctx, id)
 		if err != nil {
-			return fmt.Errorf("failed to create pubsub client: %v", err)
+			return err
 		}
-		h.decouple, err = NewMultiTopicDecoupleSink(context.Background(), WithPubsubClient(client))
+		h.decouple, err = NewMultiTopicDecoupleSink(context.Background(), WithPubsubClient(p))
 		return err
 	}
 }
 
 // MultiTopicDecoupleSinkOption is the option to configure multiTopicDecoupleSink.
-type MultiTopicDecoupleSinkOption func(sink *multiTopicDecoupleSink)
+type MultiTopicDecoupleSinkOption func(sink *multiTopicDecoupleSinkOptions)
```

Comment:
I am curious to know how is using a new `options` struct compared to functional option directly applied to the main struct. I have seen both used but not sure what's the difference.

Reply:
The options struct is used to accumulate options that aren't members of the target struct such as the pubsub client.